explainaboard/processors/processor_registry.py

```diff
@@ -3,9 +3,89 @@
+from explainaboard.processors.qa_multiple_choice import QAMultipleChoiceProcessor
+from explainaboard.processors.qa_open_domain import QAOpenDomainProcessor
+from explainaboard.processors.qa_tat import QATatProcessor
+from explainaboard.processors.summarization import SummarizationProcessor
+from explainaboard.processors.tabular_classification import (
+    TabularClassificationProcessor,
+)
+from explainaboard.processors.tabular_regression import TabularRegressionProcessor
+from explainaboard.processors.text_classification import TextClassificationProcessor
+from explainaboard.processors.text_pair_classification import (
+    TextPairClassificationProcessor,
+)
+from explainaboard.processors.word_segmentation import CWSProcessor
 
-_processor_registry: dict = {}
+
+def get_processor_class(task: TaskType) -> type[Processor]:
+    """Returns a Processor class from the given task type.
+
+    Args:
+        task: A task type.
+
+    Returns:
+        A Processor class associated with the given task type.
+
+    Raises:
+        ValueError: if the given task is not supported.
+        TypeError: if the obtained class is not a subclass of Processor class.
+    """
+    task_to_procesor = {
+        TaskType.text_classification: TextClassificationProcessor,
+        TaskType.named_entity_recognition: NERProcessor,
+        TaskType.qa_extractive: QAExtractiveProcessor,
+        TaskType.summarization: SummarizationProcessor,
+        TaskType.machine_translation: MachineTranslationProcessor,
+        TaskType.text_pair_classification: TextPairClassificationProcessor,
+        TaskType.aspect_based_sentiment_classification: AspectBasedSentimentClassificationProcessor,  # noqa: E501
+        TaskType.kg_link_tail_prediction: KGLinkTailPredictionProcessor,
+        TaskType.qa_multiple_choice: QAMultipleChoiceProcessor,
+        TaskType.qa_open_domain: QAOpenDomainProcessor,
+        TaskType.qa_tat: QATatProcessor,
+        TaskType.conditional_generation: ConditionalGenerationProcessor,
+        TaskType.word_segmentation: CWSProcessor,
+        TaskType.language_modeling: LanguageModelingProcessor,
+        TaskType.chunking: ChunkingProcessor,
+        TaskType.cloze_mutiple_choice: ClozeMultipleChoiceProcessor,
+        TaskType.cloze_generative: ClozeGenerativeProcessor,
+        TaskType.grammatical_error_correction: GrammaticalErrorCorrectionProcessor,
+        TaskType.nlg_meta_evaluation: NLGMetaEvaluationProcessor,
+        TaskType.tabular_regression: TabularRegressionProcessor,
+        TaskType.tabular_classification: TabularClassificationProcessor,
+        TaskType.argument_pair_extraction: ArgumentPairExtractionProcessor,
+    }
+    try:
+        cls = task_to_procesor[task]
+    except KeyError:
+        raise ValueError(f"{task} is not a supported task type.")
+    if not issubclass(cls, Processor):
+        raise TypeError(f"Obtained class is not a Processor: {cls.__name__}")
```

Comment:
This check is not necessary if the dict is correctly typed.

Reply:
Yes, that's better. Added a type annotation, and removed the type check.